scripts/github-actions/generate-module-index-data.js

```diff
@@ -68,157 +45,107 @@ async function getModuleDescription(
         for (const tag of tags) {
-          var description = await getModuleDescription(
+          // Using mcrModulePath because BRM module git tags do not include the modules/ prefix
+          const gitTag = `${mcrModulePath}/${tag}`;
+          const documentationUri = `https://github.com/Azure/bicep-registry-modules/tree/${gitTag}/${modulePath}/README.md`;
+          const description = await getModuleDescription(
             github,
             core,
-            path,
-            modulePath,
-            (moduleRoot = "modules"),
-            tag,
+            mainJsonPath,
+            gitTag,
             context
           );
-          if (description) {
-            properties[tag] = { description };
-            modulesWithDescriptions[modulePath] = true;
-          }
+
+          properties[tag] = { description, documentationUri };
         }
 
         moduleIndexData.push({
-          moduleName: modulePath,
+          moduleName: mcrModulePath,
           tags,
           properties,
         });
       } catch (error) {
         core.setFailed(error);
       }
     }
-  }
 
-  // AVM Resource Modules
-  for (const moduleGroup of moduleGroupsAvmRes) {
-    const moduleGroupPath = path.join("avm/res", moduleGroup);
-    const moduleNames = await getSubdirNames(fs, moduleGroupPath);
-
-    for (const moduleName of moduleNames) {
-      const modulePath = `avm/res/${moduleGroup}/${moduleName}`;
-      const moduleBicepRegistryRefSplit = modulePath
-        .split(/[\/\\]avm[\/\\]/)
-        .pop();
-      const moduleBicepRegistryRefReplace = moduleBicepRegistryRefSplit
-        .replace(/-/g, "")
-        .replace(/[\/\\]/g, "-");
-      const moduleBicepRegistryRef = moduleBicepRegistryRefReplace;
-      const versionListUrl = `https://mcr.microsoft.com/v2/bicep/${moduleBicepRegistryRef}/tags/list`;
-
-      try {
-        core.info(`Processing AVM Resource Module ${modulePath}:...`);
-        core.info(`  Retrieving AVM Resource Module ${versionListUrl}`);
-        core.info(`    Git Tag: ${modulePath}`);
-        core.info(`    BRM Ref: ${moduleBicepRegistryRef}`);
-
-        const versionListResponse = await axios.get(versionListUrl);
-        const tags = versionListResponse.data.tags.sort();
+    numberOfModuleGroupsProcessed++;
+  }
 
-        const properties = {};
-        for (const tag of tags) {
-          var description = await getModuleDescription(
-            github,
-            core,
-            path,
-            modulePath,
-            (moduleRoot = "avm/res"),
-            tag,
-            context
-          );
-          if (description) {
-            properties[tag] = { description };
-            modulesWithDescriptions[modulePath] = true;
+  for (const avmModuleRoot of ["avm/res", "avm/ptn"]) {
+    const avmModuleGroups = await getSubdirNames(fs, avmModuleRoot);
+
+    for (const moduleGroup of avmModuleGroups) {
+      const moduleGroupPath = `avm/res/${moduleGroup}`;
```

Comment:
Think we need some logic here to handle the pattern modules here too.

As in the 2 separate loops we had before handling this had the following values for:

- Resource module ```const modulePath = `avm/res/${moduleGroup}/${moduleName}`;```
- Pattern module ```const modulePath = `avm/ptn/${moduleName}`;```

Reply:
Good catch...I meant to use `avmModuleGroup` here. Will fix it.

Reply:
Done.